server/src/main/java/org/apache/cloudstack/diagnostics/RetrieveDiagnosticsServiceImpl.java

```diff
@@ -410,159 +465,204 @@ protected String getDefaultFilesForVm(String diagnosticsType, String systemVmTyp
-                    if (cmd instanceof RetrieveFilesCommand)
-                        LOGGER.error("Unable to copy " + ((RetrieveFilesCommand) cmd).getDiagnosticFileToRetrieve());
-                    else
-                        LOGGER.error("Unable to execute script " + script);
-                    return new Answer(cmd, false, "Unable to copy " + ((RetrieveFilesCommand) cmd).getDiagnosticFileToRetrieve() + " or failed to run script " + script);
-                }
-
-                if (LOGGER.isInfoEnabled()) {
-                    LOGGER.info(cmd + " on System VM " + controlIp + " completed");
+                if (inputStream != null) {
+                    inputStream.close();
                 }
-            } finally {
-                if (script != null) {
-                    SshHelper.sshExecute(controlIp, 3922, "root", getSystemVMKeyFile(), null, "rm " + script + ".log");
-                } else {
-                    SshHelper.sshExecute(controlIp, 3922, "root", getSystemVMKeyFile(), null, "rm " + ((RetrieveFilesCommand) cmd).getDiagnosticFileToRetrieve());
+                if (outputStream != null) {
+                    outputStream.close();
                 }
+            } catch (IOException ioe) {
+                LOGGER.error(" Caught exception while closing the resources");
             }
-
-            return new Answer(cmd);
-        } catch (final Throwable e) {
-            LOGGER.error("Unexpected exception: " + e.toString(), e);
-            if (cmd instanceof RetrieveFilesCommand)
-                return new Answer(cmd, false, "Failed to retrieve file " + ((RetrieveFilesCommand) cmd).getDiagnosticFileToRetrieve() + "->" + e.getMessage());
-            else
-                return new Answer(cmd, false, "Failed to execute script " + script + "->" + e.getMessage());
         }
+        return null;
     }
 
-    protected Map<String, String> retrieveDiagnosticsFiles(Long ssHostId, String diagnosticsFiles, final VMInstanceVO systemVmId, Long timeout)
-            throws ResourceUnavailableException, OperationTimedoutException, ConcurrentOperationException, JSchException, IOException {
+    private boolean copyFileFromSystemVmToServer() {
+
+        LOGGER.info("Copying the diagnostics zip file from the system vm.");
+        if (!checkForDiskSpace(vmInstance, disableThreshold)) {
+            throw new CloudRuntimeException("Unlimited disk space on primary storage.");
+        }
+        String publicIp = null;
+        //final Map<String, String> accessDetails = networkManager.getSystemVMAccessDetails(vmInstance);
+        VirtualMachine vm = vmManager.findById(vmInstance.getId());
+        final List<NicVO> nics = _nicDao.listByVmId(vm.getId());
+        for (final NicVO nic : nics) {
+            final NetworkVO network = _networksDao.findById(nic.getNetworkId());
+            final NicVO nicVO = _nicDao.findByNtwkIdAndInstanceId(network.getId(), vm.getId());
+            publicIp = nicVO.getIPv4Address();
```

Comment:
maybe break when found, no need to iterate over the rest if we are happy with an access point